src/lib/mongoBackend/mongoQueryContext.cpp

```diff
@@ -38,6 +38,268 @@
 
 /* ****************************************************************************
 *
+* someContextElementNotFound -
+*
+* Returns true if some attribut with 'found' set to 'false' is found in the CER vector passed
+* as argument
+*
+*/
+bool someContextElementNotFound(ContextElementResponseVector& cerV)
+{
+  for (unsigned int ix = 0; ix < cerV.size(); ++ix)
+  {
+    for (unsigned int jx = 0; jx < cerV.get(ix)->contextElement.contextAttributeVector.size(); ++jx)
+    {
+      if (!cerV.get(ix)->contextElement.contextAttributeVector.get(jx)->found)
+      {
+        return true;
+      }
+    }
+  }
+  return false;
+}
+
+
+/* ****************************************************************************
+*
+* searchCprForAttribute -
+*
+* Search the CPr, given the entity/attribute as argument. Actually, two CPrs can be returned
+* the "general" one at entity level or the "specific" one at attribute level
+*
+*/
+void searchCprForAttribute(EntityId& en, std::string attrName, ContextRegistrationResponseVector& crrV, std::string& perEntPa, std::string& perAttrPa)
+{
+  perEntPa  = "";
+  perAttrPa = "";
+  for (unsigned int ix = 0; ix < crrV.size(); ++ix)
+  {
+    ContextRegistrationResponse* crr = crrV.get(ix);
+    /* Is there a matching entity in the CRR? */
+    for (unsigned jx = 0; jx < crr->contextRegistration.entityIdVector.size(); ++jx)
+    {
+      EntityId* regEn = crr->contextRegistration.entityIdVector.get(jx);
+      if (regEn->id == en.id && (regEn->type == en.type || regEn->type == ""))
```

Comment:
For 'pre-conditions' is it easier to read the code if you just 'continue if not fulfilled':

    if (regEn->id != en.id || ((regEn->type != en.type) && (regEn->type != "")))
        continue;

After this, the entity matches and the loop can continue without extra blocks and indentations.
Very good 'best practise' IMHO.